crates/controller/src/mouse.rs

```diff
@@ -212,16 +212,47 @@ enum DragResolution {
     Rect(ScreenRect),
 }
 
-fn update_position(windows: Res<Windows>, mut mouse: ResMut<MousePosition>) {
+fn update_position(
+    windows: Res<Windows>,
+    hud: Query<(&GlobalTransform, &Node), With<HudVisibleNode>>,
```

Comment:
I would experiment with making this a `SystemParam` to make the API nicer. The usage would look like this:

```rust
fn update_position(
    windows: Res<Windows>,
    hud: HudNodes,
    mut mouse: ResMut<MousePosition>,
) {
...
            .filter(|position| hud.contains_point(position))
}
```